### ISSUE TITLE
Error when building the `npm` dependencies of the extensions on Windows OS

### ISSUE THREAD

**Issue by sayak-brm** [directus/api#571]
# Bug Report

## Steps to Reproduce

Build the `npm` dependencies of  the extensions on Windows OS using `npm run build` after `npm install`.

## Expected Behavior

The extensions are built without any error.

## Actual Behavior

The build stops due to the use of *nix specific shell commands.

## Other Context & Screenshots

![image](https://user-images.githubusercontent.com/5992716/47969670-19ae2600-e0a1-11e8-8a18-8a080095a57f.png)

## Technical Details

- API: Directus API 2.0.5
- OS: Windows 10 Home x64 ver 1803
- Web Server: Apache 2.4.33
- PHP Version: PHP 7.2.6
- Node.js Version: 10.13.0 (x64)
- Install Method: cloned `master` branch

## Log File

```
0 info it worked if it ends with ok
1 verbose cli [ 'D:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'D:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'clear' ]
2 info using npm@6.4.1
3 info using node@v10.13.0
4 verbose run-script [ 'preclear', 'clear', 'postclear' ]
5 info lifecycle @directus/extensions-builder@1.0.0~preclear: @directus/extensions-builder@1.0.0
6 info lifecycle @directus/extensions-builder@1.0.0~clear: @directus/extensions-builder@1.0.0
7 verbose lifecycle @directus/extensions-builder@1.0.0~clear: unsafe-perm in lifecycle true
8 verbose lifecycle @directus/extensions-builder@1.0.0~clear: PATH: D:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;D:\xampp\htdocs\api\extensions\node_modules\.bin;D:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;D:\xampp\htdocs\api\extensions\node_modules\.bin;C:\Users\user\AppData\Roaming\npm;D:\Program Files\nodejs\;C:\Python27\;C:\Python27\Scripts;C:\ProgramData\Boxstarter;D:\Program Files\ImageMagick-7.0.8-Q16;C:\Program Files (x86)\Intel\iCLS Client\;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Program Files (x86)\Common Files\Intel\Shared Libraries\redist\intel64\compiler;C:\Program Files\Intel\iCLS Client\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files\Intel\WiFi\bin\;C:\Program Files\Common Files\Intel\WirelessCommon\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;D:\Program Files\Git\cmd;D:\Program Files\PuTTY\;C:\WINDOWS\System32\OpenSSH\;D:\Program Files\Microsoft VS Code\bin;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\Intel\Intel(R) Management Engine Components\IPT;D:\xampp\php;C:\ProgramData\ComposerSetup\bin;D:\Program Files (x86)\AOMEI Backupper;D:\Program Files\nodejs\;C:\ProgramData\chocolatey\bin;D:\Users\user\AppData\Local\Programs\Python\Python36-32\Scripts\;D:\Users\user\AppData\Local\Programs\Python\Python36-32\;C:\Users\user\AppData\Local\Microsoft\WindowsApps;D:\Program Files\Microsoft VS Code\bin;C:\Users\user\AppData\Local\GitHubDesktop\bin;C:\Users\user\AppData\Local\Microsoft\WindowsApps;D:\Program Files (x86)\Nmap;C:\Users\user\AppData\Roaming\Composer\vendor\bin;C:\Users\user\AppData\Roaming\npm
9 verbose lifecycle @directus/extensions-builder@1.0.0~clear: CWD: D:\xampp\htdocs\api\extensions
10 silly lifecycle @directus/extensions-builder@1.0.0~clear: Args: [ '/d /s /c',
10 silly lifecycle   'rm -rf ../public/extensions/core/{interfaces,layouts,pages}' ]
11 silly lifecycle @directus/extensions-builder@1.0.0~clear: Returned: code: 1  signal: null
12 info lifecycle @directus/extensions-builder@1.0.0~clear: Failed to exec clear script
13 verbose stack Error: @directus/extensions-builder@1.0.0 clear: `rm -rf ../public/extensions/core/{interfaces,layouts,pages}`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (D:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\index.js:301:16)
13 verbose stack     at EventEmitter.emit (events.js:182:13)
13 verbose stack     at ChildProcess.<anonymous> (D:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:182:13)
13 verbose stack     at maybeClose (internal/child_process.js:962:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:251:5)
14 verbose pkgid @directus/extensions-builder@1.0.0
15 verbose cwd D:\xampp\htdocs\api\extensions
16 verbose Windows_NT 10.0.17134
17 verbose argv "D:\\Program Files\\nodejs\\node.exe" "D:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "clear"
18 verbose node v10.13.0
19 verbose npm  v6.4.1
20 error code ELIFECYCLE
21 error errno 1
22 error @directus/extensions-builder@1.0.0 clear: `rm -rf ../public/extensions/core/{interfaces,layouts,pages}`
22 error Exit status 1
23 error Failed at the @directus/extensions-builder@1.0.0 clear script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

**Comment by sayak-brm** [comment]
The correct way to do this in a cross-platform way is to use [rimraf][1] & [mkdirp][2].

  [1]: https://www.npmjs.com/package/rimraf
  [2]: https://www.npmjs.com/package/mkdirp

**Comment by rijkvanzanten** [comment]
Closed in #572
